embedded/src/main/java/org/jboss/as/embedded/EmbeddedStandAloneServerFactory.java

```diff
@@ -0,0 +1,479 @@
+        return standaloneServer;
+    }
+
+    static void setupCleanDirectories(Properties props) {
+        File jbossHomeDir = new File(props.getProperty(ServerEnvironment.HOME_DIR));
+        setupCleanDirectories(jbossHomeDir, props);
+    }
+
+    static void setupCleanDirectories(File jbossHomeDir, Properties props) {
+        File tempRoot = getTempRoot(props);
+        if (tempRoot == null) {
+            return;
+        }
+
+        File originalConfigDir = getFileUnderAsRoot(jbossHomeDir, props, ServerEnvironment.SERVER_CONFIG_DIR, "configuration", true);
+        File originalDataDir = getFileUnderAsRoot(jbossHomeDir, props, ServerEnvironment.SERVER_DATA_DIR, "data", false);
+
+        File configDir = new File(tempRoot, "config");
+        configDir.mkdir();
+        File dataDir = new File(tempRoot, "data");
+        dataDir.mkdir();
+        // For jboss.server.deployment.scanner.default
+        File deploymentsDir = new File(tempRoot, "deployments");
+        deploymentsDir.mkdir();
+
+        copyDirectory(originalConfigDir, configDir);
+        if (originalDataDir.exists()) {
+            copyDirectory(originalDataDir, dataDir);
+        }
+
+        props.put(ServerEnvironment.SERVER_BASE_DIR, tempRoot.getAbsolutePath());
+        props.put(ServerEnvironment.SERVER_CONFIG_DIR, configDir.getAbsolutePath());
+        props.put(ServerEnvironment.SERVER_DATA_DIR, dataDir.getAbsolutePath());
+
+    }
+
+    private static File getFileUnderAsRoot(File jbossHomeDir, Properties props, String propName, String relativeLocation, boolean mustExist) {
+        String prop = props.getProperty(propName, null);
+        if (prop == null) {
+            prop = props.getProperty(ServerEnvironment.SERVER_BASE_DIR, null);
+            if (prop == null) {
+                File dir = new File(jbossHomeDir, "standalone" + File.separator + relativeLocation);
+                if (mustExist && (!dir.exists() || !dir.isDirectory())) {
+                    throw ServerLogger.ROOT_LOGGER.embeddedServerDirectoryNotFound("standalone" + File.separator + relativeLocation, jbossHomeDir.getAbsolutePath());
+                }
+                return dir;
+            } else {
+                File server = new File(prop);
+                validateDirectory(ServerEnvironment.SERVER_BASE_DIR, server);
+                return new File(server, relativeLocation);
+            }
+        } else {
+            File dir = new File(prop);
+            validateDirectory(ServerEnvironment.SERVER_CONFIG_DIR, dir);
+            return dir;
+        }
+
+    }
+
+    private static File getTempRoot(Properties props) {
+        String tempRoot = props.getProperty(JBOSS_EMBEDDED_ROOT, null);
+        if (tempRoot == null) {
+            return null;
+        }
+
+        File root = new File(tempRoot);
+        if (!root.exists()) {
+            //Attempt to try to create the directory, in case something like target/embedded was specified
+            root.mkdirs();
```

Comment:
This is really minor, but could you use Path api here? As it will return better error message than File will. giving some context on what is going on...